tests/Command/ExpirePasswordTest.php

```diff
@@ -78,6 +82,104 @@ public function testExpirePasswordUserNotExisting() {
 		self::assertSame(67, $returnCode);
 	}
 
+	private function getAllUsers($numberOfUsers, $groupName = null) {
+		$users = [];
+		for ($i = 1; $i <= $numberOfUsers; $i++) {
+			$user = $this->createMock(IUser::class);
+			$user->expects($this->any())
+				->method('getUID')
+				->willReturn('user' . $i);
+			if ($groupName === null) {
+				$users[] = $user;
+			} else {
+				$users[$groupName][] = $user;
+			}
+		}
+		return $users;
+	}
+
+	public function testAllUsersExpirePassword() {
+		//Lets say we have 5 users
+		$users = $this->getAllUsers(5);
+
+		$this->userManager->expects($this->once())
+			->method('search')
+			->with('')
+			->willReturn($users);
+
+		$this->timeFactory
+			->method('getTime')
+			->willReturn(1530180780);
+
+		$this->config
+			->method('getAppValue')
+			->will($this->returnValueMap([
+				['password_policy', 'spv_user_password_expiration_checked', false, 'on'],
+				['password_policy', 'spv_user_password_expiration_value', 90, 10]
+			]));
+
+		$this->commandTester->execute([
+			'--all' => null
+		]);
+
+		$output = $this->commandTester->getDisplay();
+		$expectedOutput = "The password for user1 is set to expire on 2018-06-27 10:13:00 UTC.
+The password for user2 is set to expire on 2018-06-27 10:13:00 UTC.
+The password for user3 is set to expire on 2018-06-27 10:13:00 UTC.
+The password for user4 is set to expire on 2018-06-27 10:13:00 UTC.
+The password for user5 is set to expire on 2018-06-27 10:13:00 UTC.
+";
+		$this->assertEquals($expectedOutput, $output);
+	}
+
+	public function testGroupsExpirePassword() {
+		$group1Users = $this->getAllUsers(2, 'group1');
+		$group2Users = $this->getAllUsers(3, 'group2');
+		$group4Users = $this->getAllUsers(2, 'group4,foo');
+
+		$this->groupManager->expects($this->exactly(4))
+			->method('groupExists')
+			->willReturnMap([
+				['group1', true],
+				['group2', true],
+				['group3', false],
+				['group4,foo', true]
+			]);
+
+		$this->groupManager->expects($this->any())
+			->method('findUsersInGroup')
+			->willReturnOnConsecutiveCalls($group1Users['group1'], $group2Users['group2'], $group4Users['group4,foo']);
+
+		$this->timeFactory
+			->method('getTime')
+			->willReturn(1530180780);
+
+		$this->config
+			->method('getAppValue')
+			->will($this->returnValueMap([
+				['password_policy', 'spv_user_password_expiration_checked', false, 'on'],
+				['password_policy', 'spv_user_password_expiration_value', 90, 10]
+			]));
+
+		$this->commandTester->execute([
+			'--groups' => ['group1', 'group2', 'group4,foo', 'group3'],
+			'uid' => '',
+			'expiredate' => '2019-01-01 14:00:00 CET'
+		]);
+
+		$output = $this->commandTester->getDisplay();
+		$expectedOutput = "The password for user1 is set to expire on 2019-01-01 14:00:00 UTC.
+The password for user2 is set to expire on 2019-01-01 14:00:00 UTC.
+The password for user1 is set to expire on 2019-01-01 14:00:00 UTC.
```

Comment:
I guess this behavior does not hurt - if the user is a member of multiple of the groups then they get their password expired multiple times. Probably there will be multiple similar entries in the password history for the user.

Reply:
Updated the change. I would appreciate if you can have a look.

Reply:
I think we should fix this. From a UX perspective it feels wrong that the password is reset several times for the same user, specially if the command is run once. Technically, I'm not sure if  we're generating and storing useless data in the DB. In both cases people will think that it's a bug.